node/sync/src/block_sync.rs

```diff
@@ -351,20 +364,33 @@ impl<N: Network> BlockSync<N> {
 
     /// Handles the block responses from the sync pool.
     fn try_advancing_with_block_responses(&self, mut current_height: u32) {
-        while let Some(block) = self.remove_block_response(current_height + 1) {
+        while let Some(block) = self.process_next_block(current_height + 1) {
             // Ensure the block height matches.
             if block.height() != current_height + 1 {
                 warn!("Block height mismatch: expected {}, found {}", current_height + 1, block.height());
                 break;
             }
-            // Check the next block.
-            if let Err(error) = self.canon.check_next_block(&block) {
-                warn!("The next block ({}) is invalid - {error}", block.height());
-                break;
-            }
-            // Attempt to advance to the next block.
-            if let Err(error) = self.canon.advance_to_next_block(&block) {
-                warn!("{error}");
+
+            // Try to check the next block and advance to it.
+            let advanced = match self.canon.check_next_block(&block) {
+                Ok(_) => match self.canon.advance_to_next_block(&block) {
+                    Ok(_) => true,
+                    Err(err) => {
+                        warn!("{err}");
+                        false
+                    }
+                },
+                Err(err) => {
+                    warn!("{err}");
```

Comment:
note: this is a format change that might surprise users who relied on the format of [the old one](https://github.com/AleoNet/snarkOS/pull/3410/files#diff-c2e724c7020cd2ef76ea842b503a4cb85f626f19d85d60e9c685fee70e3725ddL362)

Reply:
Thanks, good find! I've added more expressive warnings similar to the previous code here: https://github.com/AleoNet/snarkOS/pull/3410/commits/d6e4f737e99361dd2ffdff6c8abf5621689545ca